common/config/uptrace.go

```diff
@@ -0,0 +1,66 @@
+package config
+
+import (
+	"errors"
+
+	"github.com/odigos-io/odigos/common"
+)
+
+const (
+	dsnKey = "UPTRACE_DSN"
+	endpointKey = "UPTRACE_ENDPOINT"
+)
+
+type Uptrace struct{}
+
+func (s *Uptrace) DestType() common.DestinationType {
+	return common.UptraceDestinationType
+}
+
+func (s *Uptrace) ModifyConfig(dest ExporterConfigurer, currentConfig *Config) error {
+	config := dest.GetConfig()
+	dsn, exists := config[dsnKey]
+	if !exists {
+		return errors.New("Uptrace url(\"UPTRACE_DSN\") not specified, gateway will not be configured for Uptrace")
+	}
+
+	endpoint, exists := config[endpointKey]
+	if !exists {
+		endpoint = "https://otlp.uptrace.dev:4317"
+	}
+
+	exporterName := "otlp/uptrace-" + dest.GetID()
+
+	currentConfig.Exporters[exporterName] = GenericMap{
+		"endpoint": endpoint,
+		"tls": GenericMap{
+			"insecure": false,
```

Comment:
If it's self hosted, wouldn't it most likely be used with insecure connection?

Reply:
I think Uptrace is mostly used as a service. That's what I've always done at least.

I will add a check for endpoint - if it's http, will make this insecure.